packages/backend-plugin-api/src/services/definitions/AuditorService.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ErrorLike } from '@backstage/errors';
+import type { JsonObject } from '@backstage/types';
+import type { Request } from 'express';
+
+/**
+ * Indicates the event was successful.
+ *
+ * @public
+ */
+export type AuditorEventSuccessStatus = { status: 'succeeded' };
+
+/**
+ * Indicates the event failed and includes details about the encountered errors.
+ *
+ * @public
+ */
+export type AuditorEventFailureStatus<E = ErrorLike> = {
+  status: 'failed';
+  errors: E[];
+};
+
+/**
+ * Indicates the event is unknown.
+ *
+ * @public
+ */
+export type AuditorEventUnknownStatus = { status: 'unknown' };
+
+/**
+ * @public
+ */
+export type AuditorEventStatus =
+  | AuditorEventSuccessStatus
+  | AuditorEventFailureStatus
+  | AuditorEventUnknownStatus;
+
+/**
+ * Options for creating an auditor event.
+ *
+ * @public
+ */
+export type AuditorEventOptions<T extends JsonObject> = {
+  /**
+   * The name of the audit event, formatted in PascalCase (e.g., "UserLogin", "FileDownload").
+   *
+   * If applicable, it's required to prefix the `eventName` with the name of the component
+   * responsible for the event (e.g., "ScaffolderTaskRead", "CatalogEntityFetch").
+   * This improves searchability within the central log collector.
+   */
+  eventName: string;
```

Comment:
I think `eventId` fits better here imo, since this is much more of an identifier.